### DISCUSSION TITLE
Resource HasMany Create Return 404

### DISCUSSION THREAD

**Discussion by Patryk48** [laravel/nova-issues#5772] [Q&A]
- Laravel Version: 10.10.0
- Nova Version: 4.0.0
- PHP Version: 8.1.0
- Operating System and Version: Ubumtu 21.04
- Browser type and version: Chrome
- Reproduction Repository:

### Description:
When create has many relation resource, and click create button in panel relation return 404

### Detailed steps to reproduce the issue on a fresh Nova installation:

Create Relation HasMany in Model
Use HasMany field in Resource
Go to view resource and click relation
![2023-08-03_16h19_57](https://github.com/laravel/nova-issues/assets/138042563/b2ddc4b5-05d2-40d4-b7ee-6737b5cf226c)
![2023-08-03_16h20_13](https://github.com/laravel/nova-issues/assets/138042563/ce4655a3-2ea7-4c90-bf8a-e84528b91fd5)
![2023-08-03_16h22_37](https://github.com/laravel/nova-issues/assets/138042563/1c0765b4-c18e-4839-b781-b673c61b29a6)

**Comment by davidhemphill** [comment]
This was converted to a discussion because you didn't provide a reproduction repository or the complete code to demonstrate the issue. 

Additionally, since this is a well-tested feature, so it's likely a configuration issue.

**Reply by mlordi** [reply]
In my case I was getting the same error and it turns out that I was using the depends on feature on my resource and it didn't know how to populate the field with the id that was passed in from the URL. I had to update my code to search for the id or return null if it didn't exist.

```
BelongsTo::make('Project')->withoutTrashed()->sortable()->nullable()->hideFromIndex()
                ->dependsOn(['customer'], function (BelongsTo $field, NovaRequest $request, FormData $data) {
                    if($data->customer != null) {
                        $field->relatableQueryUsing(function ($request, $query) use ($data) {
                            return $query->where('user_id', $data->customer);
                        });
                    }
                    else {
                        return '';
                    }
                }),
```
